File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
     int slow=n,fast=n;
        do {
            slow=getSquareSum(slow);
            fast=getSquareSum(getSquareSum(fast));
        }while (slow!=fast);
        return slow==1;

    }

    private int getSquareSum(int num) {
        int sum=0,digit;
        while(num>0){
            digit=num%10;
            sum+=digit*digit;
            num/=10;
        }
        return sum;
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def numSquareSum(n):
            squareSum = 0
            while(n):
                squareSum += (n % 10) * (n % 10)
                n = int(n / 10)
            return squareSum
        sl=fs=n
        while True:
            sl=numSquareSum(sl)
            fs=numSquareSum(numSquareSum(fs))
            if sl!=fs:continue
            else: break
        return sl==1
        